Sources/PackageModel/Manifest/TargetDescription.swift

```diff
@@ -22,7 +22,7 @@ public struct TargetDescription: Equatable, Codable {
     }
 
     /// Represents a target's dependency on another entity.
-    public enum Dependency: Equatable {
+    public enum Dependency: Equatable, Hashable {
```

Comment:
Relying on the synthesized `Hashable` like this will not catch the following sort of duplicate:

```swift
"a", // i.e. .byName(name: "a")
.target(name: "a")
```

On the other hand, comparing only the name will snag valid pairs, such as this:

```swift
.target(name: "a"),
.product(name: "a", package: "c")
```

Reply:
Does `byName` always represent a target? If so, it might be easiest to convert one to the other for the purposes of this comparison 🤔

Reply:
Does `byName` always represent a target? If so, it might be easiest to convert one to the other for this comparison 🤔

Reply:
No. It is short for either `.target(name: "a")` or `.product(name: "a", package: "a"`), or in pre‐5.2 tools-versions, `.product(name: "a", package: /* all */)`. Figuring it out either requires more elaborate logic, or performing the check later in the loading process after these sorts of connections have already been resolved.

Reply:
> either requires more elaborate logic

Would this not just duplicate the resolution process that takes place later?

> or performing the check later in the loading process

This makes more sense to me.

What is your opinion on this?